samples/README.MD

```diff
@@ -3,43 +3,107 @@
 Developers like to learn by looking at code, and so the Azure SDK comes with a myriad of code samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.
 
 ## Structure of the Repository
+
 The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.
 
-`/samples` (this folder)<br>
-&nbsp;&nbsp;&nbsp;&nbsp;`README.md` (this file)<br>
-`/sdk` (folder containing sources for all SDK packages)<br>
-&nbsp;&nbsp;&nbsp;&nbsp;`/<service>` (e.g. storage)<br>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/<package>` (e.g. Azure.Storage.Blobs)<br>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`README.md`** (package READMEs contain hello world samples)<br>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;**`/samples`** (package-specific samples)<br>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/src`<br>
-&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`/tests`<br>
-
-##  Getting Started (a.k.a. `Hello World`) Samples
+`/samples` (this folder)\
```

Comment:
This looks quite similar to what we have in the DPG library author docs: https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/Create_Samples_Guidance.md

Would it make sense for these docs to link to each other?  How can we make sure they are both updated if one is updated?

@jsquire's point about Samples guidance in the CONTRIBUTING.md (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets) is good as well ... it looks like we've got samples guidance in at least four places now.  Would it be worth picking a single place to discuss all of the different concepts and then linking to the central guidance from each of the necessary places?